### ISSUE TITLE
Multi labeled Image classification - Help Needed

### ISSUE THREAD

**Issue by Padalingam** [apple/turicreate#287]
Hi All,
when i tried to predict two labeled image classification, its working fine. when i tried more than two image classification its not predicting correct value

Correct Prediction is 
data["animalType"] = data["path"].apply(lambda path: "Cat" if "cat" in path else "Dog")
Wrong Prediction is
data["animalType"] = data["path"].apply(lambda path: "Cat" if "cat" in path else ("Dog" if "dog" in path else "lion"))

I have three different dataset on the respective places

**Comment by TobyRoseman** [comment]
@Padalingam - We are going to need a lot more information to in order to help you. For starters, what accuracy are you getting with three classes vs with two classes? How many examples do you have in each category?

**Comment by Padalingam** [comment]
@TobyRoseman I have 30 images in each category. After converting the core ml model, if i feed any one of three images as a input, i am getting a single category in three classes as a output and its accuracy is vary between 0.5 to 1.0

Example 1: if i feed cat image, i am getting dog 
Example2: If i feed dog image, i am getting dog
Example3: If i feed lion image, i am getting dog

But, if i have two category classes this works fine and it predicts perfectly.

What i am looking is some thing similar to Resnet core ml model. My requirement is to identify the different chairs and its parts. I might n number of chairs and its n number of parts

**Comment by TobyRoseman** [comment]
@Padalingam -

1 - ResNet is used to get what are called deep features. You can think of it as basically a dimensionality reduction from the input, where the input is the RGB values for each pixel. A logistic regression is then trained on these deep features. 
For details, see: https://github.com/apple/turicreate/blob/master/userguide/image_classifier/how-it-works.md
The actual code is [here](https://github.com/apple/turicreate/blob/ca62331a2161332a2e49fca1d6926325cc71519b/src/unity/python/turicreate/toolkits/image_classifier/image_classifier.py#L115).

2 - Yes, I think so. Recognizing different parts of a chair certainly seems like an Image classification task.

3 - I think you are going to need more data (i.e. images to train on). I've had success training on only 30 examples per category but that was for categories that are visually distinct from each other. Given how similar different parts of a chair look, I think you're going to need much more than 30 pictures per category.

**Comment by Padalingam** [comment]
Thanks so much and sounds good. One more question, do i need to maintain equal number images in each category? 

What happen, if i have 25,20, and 30 images in three labeled image classification? Will it predict correctly?

**Comment by TobyRoseman** [comment]
@Padalingam - good question. As long as your class sizes are roughly equal you should be fine. You do not need to have exactly the same number of each category in your training data.

**Comment by Padalingam** [comment]
Thanks. Whether deep learning supports in turicreate? If so, how can i implement it?
or If i specify my model as resnet-50 then, the classifier will belongs to deep learning?

**Comment by TobyRoseman** [comment]
@Padalingam - Sorry, I do not understand your questions. Image classification uses deep learning (either a trained resnet-50 neural network or a trained squeezenet neural network) to generate the deep features. If that doesn't answer your questions, please clarify your questions.

**Comment by Padalingam** [comment]
Yes, my question is how can i achieve deep learning in turicreate(Image classification)
